py-polars/tests/unit/namespaces/test_datetime.py

```diff
@@ -1009,62 +1021,91 @@ def test_mean(
     s = pl.Series(values)
     assert s.dt.mean() == expected_mean
 
-    if s.dtype == pl.Datetime:
+    if s.dtype in (pl.Datetime, pl.Duration, pl.Time):
         assert s.mean() == expected_mean
 
 
 @pytest.mark.parametrize(
     ("values", "expected_mean"),
     [
+        ([None], None),
         (
             [datetime(2022, 1, 1), datetime(2022, 1, 2), datetime(2024, 5, 15)],
             datetime(2022, 10, 16, 16, 0, 0),
         ),
     ],
-    ids=["spread_skewed_dt"],
+    ids=["None_dt", "spread_skewed_dt"],
 )
-def test_datetime_mean_with_tu(values: list[datetime], expected_mean: datetime) -> None:
-    assert pl.Series(values, dtype=pl.Duration("ms")).mean() == expected_mean
-    assert pl.Series(values, dtype=pl.Duration("ms")).dt.mean() == expected_mean
-    assert pl.Series(values, dtype=pl.Duration("us")).mean() == expected_mean
-    assert pl.Series(values, dtype=pl.Duration("us")).dt.mean() == expected_mean
-    assert pl.Series(values, dtype=pl.Duration("ns")).mean() == expected_mean
-    assert pl.Series(values, dtype=pl.Duration("ns")).dt.mean() == expected_mean
+@pytest.mark.parametrize("time_unit", ["ms", "us", "ns"])
+def test_datetime_mean_with_tu(
+    values: list[datetime], expected_mean: datetime, time_unit: str
+) -> None:
+    assert pl.Series(values, dtype=pl.Duration(time_unit)).mean() == expected_mean  # type:ignore[arg-type]
+    assert pl.Series(values, dtype=pl.Duration(time_unit)).dt.mean() == expected_mean  # type:ignore[arg-type]
```

Comment:
I think you can use `time:unit: TimeUnit`, and then you don't need the type: ignores

Reply:
Much better, thanks!